draft-ietf-tls-esni.md

```diff
@@ -772,6 +778,11 @@ the TLS connection to the backend server (if in Split Mode). In
 the latter case, it does not make any changes to the TLS
 messages, but just blindly forwards them.
 
+If the ClientHello is the result of a HelloRetryRequest, servers MUST
+abort the connection with an "illegal_parameter" alert if the decrypted
+ClientESNIInner from the second ClientHello does not match that
+of the first ClientHello.
```

Comment:
I agree that servers MUST verify the value of the ESNI extension, but the fields being required to be validated seems to differ from the the fields that are required to be consistent. Is there a reason for the two to be different?

FWIW, "Handling the server response" section states as follows:
> If the server sends a HelloRetryRequest in response to the ClientHello and the client can send a second updated ClientHello per the rules in {{RFC8446}}, the "encrypted_server_name" extension values which do not depend on the (possibly updated) KeyShareClientHello, i.e,, ClientEncryptedSNI.suite, ClientEncryptedSNI.key_share, and ClientEncryptedSNI.record_digest, MUST NOT change across ClientHello messages. Moreover, ClientESNIInner.nonce and ClientESNIInner.realSNI MUST not change across ClientHello messages. Informally, the values of all unencrypted extension information, as well as the inner extension plaintext, must be consistent between the first and second ClientHello messages.

Reply:
No reason -- just an oversight on my part. It should be fixed now!